pynestml/codegeneration/nest_code_generator.py

```diff
@@ -138,6 +161,12 @@ def raise_helper(self, msg):
         raise TemplateRuntimeError(msg)
 
     def set_options(self, options: Mapping[str, Any]) -> Mapping[str, Any]:
+        # insist on using the old Archiving_Node class for NEST 2
+        if self.get_option("nest_version").startswith("v2"):
```

Comment:
Is this required here again? It is already mentioned in line number 120.

Reply:
I agree it's not ideal, but it's duplicated to keep things consistent. It might be that it's set by the constructor, but overwritten by the ``generate_code()`` invocation with custom code generator options (e.g. setting ``StructuralPlasticityNode`` as the parent class), which needs to be overridden again when the actual code is generated.